File: Java/Migration/unit-tests-java/src/test/java/p04_mocks/Ex05ZeroInteractions.java
package p04_mocks;

import org.testng.annotations.Test;
import p04_mockito.NullCommand;
import p04_mockito.SharedCore;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verifyZeroInteractions;

public class Ex05ZeroInteractions {
  @Test
  public void shouldDoNothingWhenExecuted() {
    //GIVEN
    NullCommand command = new NullCommand();

    //WHEN
    SharedCore core = mock(SharedCore.class);
    command.ExecuteOn(core);

    //THEN
    verifyZeroInteractions(core /* core, core */);
  }


}

File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P04NSubstitute/Ex05ZeroInteractions.cs
using NSubstitute;
using NUnit.Framework;
using TddEbook.TddToolkit.NSubstitute;

namespace unit_tests_csharp.P04NSubstitute
{
  public class Ex05ZeroInteractions
  {
    [Test]
    public void ShouldDoNothingWhenExecuted()
    {
      //GIVEN
      var command = new NullCommand();

      //WHEN
      var core = Substitute.For<ISharedCore>();
      command.ExecuteOn(core);

      //THEN
      core.ReceivedNothing();
    }
  }
}